ranger-core/src/main/java/io/appform/ranger/core/model/ServiceRegistry.java

```diff
@@ -15,15 +15,22 @@
  */
 package io.appform.ranger.core.model;
 
+import java.util.concurrent.atomic.AtomicBoolean;
 import lombok.Data;
 
 import java.util.List;
 
 @Data
 public abstract class ServiceRegistry<T> {
     private final Service service;
+    private final AtomicBoolean initialRefreshCompleted = new AtomicBoolean(false);
 
     public abstract List<ServiceNode<T>> nodeList();
 
-    public abstract void updateNodes(List<ServiceNode<T>> nodes);
+    public abstract void update(List<ServiceNode<T>> nodes);
```

Comment:
You do not want people to call this directly. Make this `protected`

Reply:
Done